contrib/grafana-watcher/grafana/grafana.go

```diff
@@ -43,3 +44,15 @@ func (c *Clientset) Dashboards() DashboardsInterface {
 func (c *Clientset) Datasources() DatasourcesInterface {
 	return NewDatasourcesClient(c.BaseUrl, c.HTTPClient)
 }
+
+func doRequest(c *http.Client, req *http.Request) error {
+	resp, err := c.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("doRequest failed, Code:%d, Status:%s", resp.StatusCode, resp.Status)
```

Comment:
Thanks for the comment, checked the grafana [dashboard](http://docs.grafana.org/http_api/dashboard/) and [datasource](http://docs.grafana.org/http_api/data_source/) http api docs, http.StatusOK is the only valid success code.